### DISCUSSION TITLE
save range date variable

### DISCUSSION THREAD

**Discussion by bderkk** [zauberzeug/nicegui#332] [Q&A] [answered]
Hi! I'm new with nicegui and I'd like to port my code and use it. I'm having problems to save the range of dates when they are selected. I would like to save "from" date, to "x" variable, and "to" date, to "y" variable for exemple. Sorry if it is a silly question...

Thank you!

**Comment by falkoschindler** [comment]
Your question might be related to #268 or #299 where we've also been asked about storing information from events into variables or other global state. If this is not what you've been looking for, you might want to provide a bit more context: What have you tried already and what exactly does not work?

**Reply by bderkk** [reply]
Hi! Thank you for the answer. Well, for exemple it's an app where you choose a range of dates, then the app ask you for how many hours have you work in each day inside the range, so I need to now if it's a weekday or not, calculate the price of the extra hours, and save it to export it after to a table, csv or pdf. I've got a functional app using tkinter, but I love how it looks NiceGui and the other features like Auth, it's web based (I'd not need to make and executable and I could use it anywhere...) Thank you for your help, and sorry, I'm pretty new with all of this... :)

EDIT: I try ui.date(), and I'm able to extract and specific date, but when it's a range, I can not extract the dates, for exemple if I select 01/02/2023 to 03/02/2023, I need to store both dates, for exemple: x="01/02/2023" and y="03/02/2023"

**Comment by falkoschindler** [comment, marked as answer]
How about this:

```py
x = None
y = None

def handle_date_range(event) -> None:
    global x, y
    x = event.value['from']
    y = event.value['to']
    print(x, y, flush=True)

ui.date(on_change=handle_date_range).props('range')
```

**Reply by bderkk** [reply]
Ohh! Perfect!, actually I was trying something like that (not same) and it wasn't working but your exemple is working perfect, thank you very much!:

```python
from datetime import datetime
from nicegui import ui

start_date = None
end_date = None

def handle_change(value):
    global start_date, end_date
    start_date = value['from']
    end_date = value['to']

    print(start_date, end_date)

ui.date({'from': '2023-01-01', 'to': '2023-01-05'}, on_change=handle_change).props('range')

ui.run()
```
